src/SprykerEco/Zed/Unzer/Business/Quote/UnzerKeypairQuoteExpander.php

```
@@ -0,0 +1,130 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\Unzer\Business\Quote;
+
+use Generated\Shared\Transfer\QuoteTransfer;
+use Generated\Shared\Transfer\UnzerCredentialsConditionsTransfer;
+use Generated\Shared\Transfer\UnzerCredentialsCriteriaTransfer;
+use SprykerEco\Shared\Unzer\UnzerConstants;
+use SprykerEco\Zed\Unzer\Business\Credentials\UnzerCredentialsResolverInterface;
+
+class UnzerKeypairQuoteExpander implements UnzerKeypairQuoteExpanderInterface
+{
+    /**
+     * @var \SprykerEco\Zed\Unzer\Business\Credentials\UnzerCredentialsResolverInterface
+     */
+    protected $unzerCredentialsResolver;
+
+    /**
+     * @param \SprykerEco\Zed\Unzer\Business\Credentials\UnzerCredentialsResolverInterface $unzerCredentialsResolver
+     */
+    public function __construct(UnzerCredentialsResolverInterface $unzerCredentialsResolver)
+    {
+        $this->unzerCredentialsResolver = $unzerCredentialsResolver;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return \Generated\Shared\Transfer\QuoteTransfer
+     */
+    public function expandQuoteWithUnzerKeypair(QuoteTransfer $quoteTransfer): QuoteTransfer
+    {
+        $uniqueMerchantReferences = $this->extractUniqueMerchantReferences($quoteTransfer);
+
+        if ($quoteTransfer->getPaymentOrFail()->getUnzerPaymentOrFail()->getIsMarketplaceOrFail()) {
```

Does it make sense to call `getIsMarketplaceOrFail()` or we should throw an exception in case of `null`?

---

I suggest to leave as is because it will fail only if our customer breaks integration steps of this module